apps/emqx/src/emqx_types.erl

```diff
@@ -163,7 +164,7 @@
     dn => binary(),
     atom() => term()
 }.
--type clientid() :: binary() | atom().
+-type tenant() :: maybe(binary()).
```

Comment:
use `<<>>` fo no tenant.
`undefined` is 9 bytes when serialised to external format.
```suggestion
-type tenant() :: binary().
```